bundles/GraphicServiceBilling/Controller/MainController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of aakb/jira_economics.
+ *
+ * (c) 2019 ITK Development
+ *
+ * This source file is subject to the MIT license.
+ */
+
+namespace GraphicServiceBilling\Controller;
+
+use App\Service\MenuService;
+use GraphicServiceBilling\Service\GraphicServiceBillingService;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\DateTimeType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * Class MainController.
+ *
+ * @Route("", name="graphic_service_billing_")
+ */
+class MainController extends AbstractController
+{
+    /**
+     * @Route("", name="index")
+     *
+     * @param \Symfony\Component\HttpFoundation\Request                   $request
+     * @param \App\Service\MenuService                                    $menuService
+     * @param \GraphicServiceBilling\Service\GraphicServiceBillingService $graphicServiceBillingService
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \PhpOffice\PhpSpreadsheet\Exception
+     * @throws \PhpOffice\PhpSpreadsheet\Writer\Exception
+     */
+    public function index(Request $request, MenuService $menuService, GraphicServiceBillingService $graphicServiceBillingService)
+    {
+        $startOfWeek = (new \DateTime(date('c', strtotime('this week', time()))))->setTime(0, 0);
+        $endOfWeek = (new \DateTime($startOfWeek->format('c')))->add(new \DateInterval('P6D'))->setTime(23, 59);
+
+        $formBuilder = $this->createFormBuilder();
+        $formBuilder->add('from', DateTimeType::class, [
+            'label' => 'gs_billing_form.from',
+            'data' => $startOfWeek,
+            'date_widget' => 'single_text',
+        ]);
+        $formBuilder->add('to', DateTimeType::class, [
+            'label' => 'gs_billing_form.to',
+            'data' => $endOfWeek,
+            'date_widget' => 'single_text',
+        ]);
+        $formBuilder->add('marketing', CheckboxType::class, [
+            'label' => 'gs_billing_form.marketing_account',
+            'required' => false,
+        ]);
+        $formBuilder->add('submit', SubmitType::class, [
+            'label' => 'gs_billing_form.preview',
+        ]);
+        $formBuilder->add('download', SubmitType::class, [
+            'label' => 'gs_billing_form.download',
+            'attr' => [
+                'class' => 'btn-secondary',
+            ],
+        ]);
+
+        $form = $formBuilder->getForm();
+
+        $form->handleRequest($request);
+
+        $preview = null;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $download = $form->get('download')->isClicked();
```

Comment:
Some would argue that this should be handled by a service. Others would argue that a controller _is_ a service. I'm not going any of these places …

Reply:
Added here since it has to do with the response, but some of the code could be moved to a service.